support-metrics-client/src/main/java/io/confluent/support/metrics/SupportedKafka.java

```diff
@@ -48,8 +45,15 @@ public static void main(String[] args) throws Exception {
       final SupportedServerStartable supportedServerStartable =
           new SupportedServerStartable(serverProps);
 
-      // register signal handler to log termination due to SIGTERM, SIGHUP and SIGINT (control-c)
-      registerLoggingSignalHandler();
+      if (!isWindows() && !isIbmJdk()) {
```

Comment:
@ijuma could we have `.register()` throw an `UnsupportedVm / UsupportedOSException` to handle this cleanly?

Reply:
ok, just have it return true/false then?

Reply:
The problem with `true`/`false` is that it doesn't provide any context, so it's a bit of an anti-pattern. In any case, the AK method we're using here is `internal` and this is a short-term solution until there's a better way.